### PR TITLE
feat: add trainer config in scheduler

### PR DESCRIPTION
**Pull request by MinH-09**
## Description
Scheduler add trainer configuration

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: scheduler/config/config.go [hunk 1]
```diff
@@ -69,6 +69,9 @@ type Config struct {
 
 	// NetworkTopology configuration.
 	NetworkTopology NetworkTopologyConfig `yaml:"networkTopology" mapstructure:"networkTopology"`
+
+	// Trainer configuration
```
Comment (zhoulizhen) [inline scheduler/config/config.go:73]: // Trainer configuration.

File: scheduler/config/config.go [hunk 2]
```diff
@@ -320,6 +323,42 @@ type ProbeConfig struct {
 	SyncCount int `mapstructure:"syncCount" yaml:"syncCount"`
 }
 
+type TrainerConfig struct {
+	// Enable trainer service.
+	Enable bool `yaml:"enable" mapstructure:"enable"`
+
+	// Addr is trainer service address
```
Comment (zhoulizhen) [inline scheduler/config/config.go:330]: // Addr is trainer service address.

File: scheduler/config/constants.go [hunk 3]
```diff
@@ -171,3 +171,25 @@ const (
 	// DefaultProbeSyncCount is the default number of probing hosts.
 	DefaultProbeSyncCount = 50
 )
+
+const (
+	// DefaultNetworkRecordDataPath is default data path for network record
```
Comment (zhoulizhen) [inline scheduler/config/constants.go:176]: // DefaultNetworkRecordDataPath is default data path for network record.

File: scheduler/config/constants.go [hunk 4]
```diff
@@ -171,3 +171,25 @@ const (
...
+	DefaultNetworkRecordDataPath = "networkRecord"
+
+	// DefaultStorageMaxSize is the default maximum size of network record.
```
Comment (zhoulizhen) [inline scheduler/config/constants.go:179]: // DefaultNetworkRecordMaxSize is the default maximum size of network record.

File: scheduler/config/constants.go [hunk 5]
```diff
@@ -171,3 +171,25 @@ const (
...
+	DefaultNetworkRecordMaxSize = 100
+
+	// The default setting needs to be changed after testing.
+	// DefaultMaxBackups is the default maximum count of network records.
```
Comment (zhoulizhen) [inline scheduler/config/constants.go:183]: // DefaultNetworkRecordMaxBackups is the default maximum count of network record backups.

File: scheduler/config/constants.go [hunk 6]
```diff
@@ -171,3 +171,25 @@ const (
...
+	DefaultNetworkRecordMaxBackups = 10
+
+	// DefaultNetworkRecordDataPath is default data path for network record
+	DefaultHistoricalRecordDataPath = "historicalRecord"
```
Comment (zhoulizhen) [inline scheduler/config/constants.go:187]: // DefaultHistoricalRecordDataPath is default data path for network historical record.

File: scheduler/config/constants.go [hunk 7]
```diff
@@ -171,3 +171,25 @@ const (
...
+
+	// DefaultHistoricalRecordMaxSize is the default maximum size of network record.
```
Comment (zhoulizhen) [inline scheduler/config/constants.go:189]: // DefaultHistoricalRecordMaxSize is the default maximum size of network historical record.

File: scheduler/config/constants.go [hunk 8]
```diff
@@ -171,3 +171,25 @@ const (
...
+	DefaultHistoricalRecordMaxSize = 100
+
+	// The default setting needs to be changed after testing.
```
Comment (zhoulizhen) [inline scheduler/config/constants.go:192]: Add TODO（HM）

File: scheduler/config/constants.go [hunk 9]
```diff
@@ -171,3 +171,25 @@ const (
...
+	// DefaultHistoricalRecordMaxBackups is the default maximum count of network records.
```
Comment (zhoulizhen) [inline scheduler/config/constants.go:193]: // DefaultHistoricalRecordMaxBackups is the default maximum count of network historical record backups.

File: scheduler/config/constants.go [hunk 10]
```diff
@@ -171,3 +171,25 @@ const (
...
 )
+
+const (
+	// DefaultNetworkRecordDataPath is default data path for network record
+	DefaultNetworkRecordDataPath = "networkRecord"
+
+	// DefaultStorageMaxSize is the default maximum size of network record.
+	DefaultNetworkRecordMaxSize = 100
+
+	// The default setting needs to be changed after testing.
```
Comment (zhoulizhen) [inline scheduler/config/constants.go:182]: Add TODO(HM)

**Review by gaius-qi — APPROVED** [review]
LGTM

**Pull request merged**